core/data_operations/account_data.py

```diff
@@ -28,14 +32,65 @@ def check_account_exists_then_return(
             return {
                 "account_id": account.id,
                 "email_address": account.email,
-            }
+                "roles": account.serialize["roles"],
+            }, 200
         else:
             return account
     except sqlalchemy.exc.NoResultFound:
         return NoContent, 404
 
 
-def get_account_data_by_email(email: str, as_json: bool = True) -> Tuple[dict, int]:
+def update_account(account_id: str, roles: List[str]) -> Tuple[dict, int]:
+    """
+    Updates an account rols
+    Args:
+        account_id (str): The account id
+        roles (List[str]): A list of roles to set
+
+    Returns:
+        A tuple with account status as json and a status code.
+
+    """
+    # Check account exists
+    try:
+        account = (
+            db.session.query(Account).filter(Account.id == account_id).one()
+        )
+    except sqlalchemy.exc.NoResultFound:
+        return NoContent, 404
+    # Check all roles are valid
+    for role in roles:
+        try:
+            RoleType[role.upper()]
+        except KeyError:
+            return {"error": f"Role '{role}' is not valid"}, 401
```

Comment:
📓 The db will do this validation for you when you pass in the capitalised role.

Reply:
So, I went with this approach so as to check validation on all the items before doing any updates (i.e. if one fails they all fail). I guess I could do a session.commit and rollback if it fails... or you think this ok?